File: file_processing/pl/lte_stdin_threshold.pl
#!/usr/bin/perl
## prints numbers from STDIN that are <= $ARGV[0]
## to run: more numbers.txt | perl lte_input_thresh.pl 10
use strict;
use warnings;
my $thresh = shift();
while ( <> ) { print $_ if ( $_ <= $thresh ); }

File: file_processing/py/lte_stdin_threshold.py
#!/usr/bin/python
## print integers from STDIN that are <= sys.argv[1]
## to run: more numbers.txt | python lte_input_thresh.py 10
import sys
thresh = int(sys.argv[1])
for n in [int(x) for x in sys.stdin.readlines() if int(x) <= thresh]:
    print n
